include/dirac_quda.h

```diff
@@ -2238,6 +2250,69 @@ namespace quda {
     }
   };
 
+  /**
+     Gloms onto a DiracMatrix and provides an operator() for its G5M method
+  */
+  class DiracG5M : public DiracMatrix
+  {
+
+  public:
+    DiracG5M(const Dirac &d) : DiracMatrix(d) { }
+    DiracG5M(const Dirac *d) : DiracMatrix(d) { }
+
+    void operator()(ColorSpinorField &out, const ColorSpinorField &in) const
+    {
+      dirac->M(out, in);
+      if (shift != 0.0) blas::axpy(shift, const_cast<ColorSpinorField &>(in), out);
+      gamma5(out, out);
```

Comment:
If you want to extend this to staggered, this should work---staggered gamma5 corresponds to flipping the sign on odd sites. (I can test this later if you pop it in, or I can pop it in myself and test it, or whatever works best!)
```
if (isStaggered()) {
  if (out.Subset() == QUDA_FULL_SITE_SUBSET) {
    blas::ax(-1.0, out.Odd());
  } else if (out.Subset() == QUDA_PARITY_SITE_SUBSET && out.SuggestedParity() == QUDA_ODD_PARITY) {
    // harmless if suggested parity is unset
    blas::ax(-1.0, out.Odd());
  }
}
```

Reply:
You best put that in, I'm not even sure at which line to place it!